Add composite action to push images to ECR

wchrisjohnson opened the pull request:
Add composite action to push images to ECR.

wchrisjohnson committed:
Add ECR push with multi-arch support

- Add composite action for pushing to ECR with OIDC auth
- Support multi-arch builds (amd64, arm64)
- Add ECR push step after Runtime Registry push
- Uses existing ARCHITECTURES secret

wchrisjohnson committed:
Add feature branch to workflow trigger for testing

wchrisjohnson committed:
Add QEMU and buildx setup for multi-arch builds

Ubuntu 24.04 runners require explicit QEMU setup for cross-platform
builds. This is now best practice for multi-arch Docker builds.

wchrisjohnson committed:
Fix bash array expansion in manifest creation

wchrisjohnson committed:
Remove github context from action description

ryanbrainard commented on .github/actions/push-to-ecr/action.yml:
```diff
@@ -0,0 +1,130 @@
+    description: 'Image tag to use (e.g., commit SHA or version number)'
+    required: true
+  image-name:
+    description: 'Local Docker image name to push (defaults to ecr-repository:image-tag)'
```
is this true? it doesn't seem to be implemented this way and `image-name` isn't passed in, so will default to the `ARCH`.

wchrisjohnson committed:
Remove unused image-name parameter from push-to-ecr action

- Remove image-name input parameter that was not being used
- Simplify push logic to use expected image naming conventions:
  - Single-arch: repository name with tag
  - Multi-arch: architecture name directly (e.g., amd64, arm64)
- Remove conditional logic for LOCAL_IMAGE variable

ntlam commented on .github/workflows/docker.yml:
```diff
@@ -3,12 +3,24 @@ on:
   push:
     branches:
       - "master"
+      - "add-ecr-push"
```
nit: we should remove this before merging

ntlam commented on .github/actions/push-to-ecr/action.yml:
```diff
@@ -0,0 +1,119 @@
+    description: 'AWS region for ECR'
+    required: false
+    default: 'us-east-1'
+  aws-role-arn:
```
maybe this is unwarranted paranoia (I don't think this is sensitive and all that useful in a vacuum) but I wonder if we should remove the default and make this a repo secret as not to reveal too much information 🤷

wchrisjohnson replied:
I don't think I so. Given it's a public repo, I agree. 

On second thought, even better @ntlam would be to move this action out if the repo altogether and reference it from a repo.

wchrisjohnson committed:
Refactor push-to-ecr action for clarity

- Rename old action to push-to-ecr-v0 for reference
- Create new simplified push-to-ecr action
- Rename inputs for clarity:
  - ecr-repository → ecr-repository-path
  - architectures → source-image-tags
- Simplify validation to only enforce s/ prefix and GitHub repo name
- Remove verbose logging and emoji from push logic
- Update docker.yml workflow to use new action inputs

wchrisjohnson commented:
Closing this PR due to git worktree being out of sync.

Repacfed by #65
